gapic/schema/wrappers.py

```diff
@@ -383,14 +395,59 @@ def oneof_fields(self, include_optional=False):
 
         return oneof_fields
 
+    @utils.cached_property
+    def extended_operation_request_fields(self) -> Sequence[Field]:
+        """
+        If this message is the request for a method that uses extended operations,
+        return the fields that correspond to operation request fields in the operation message.
+        """
+        return tuple(
+            f
+            for f in self.fields.values()
+            if f.operation_request_field
+        )
+
+    @utils.cached_property
+    def extended_operation_response_fields(self) -> Sequence[Field]:
+        """
+        If this message is the request for a method that uses extended operations,
+        return the fields that correspond to operation response fields in the polling message.
+        """
+        return tuple(
+            f
+            for f in self.fields.values()
+            if f.operation_response_field
+        )
+
+    @utils.cached_property
+    def differently_named_extended_operation_fields(self) -> Optional[Dict[str, Field]]:
+        if not self.is_extended_operation:
+            return None
+
+        def canonical_name(field):
+            return OperationResponseMapping.Name(field.operation_field).lower()
+
+        OperationResponseMapping = ex_ops_pb2.OperationResponseMapping
+        default_field_names = [
+            k.lower()
+            # The first variant is UNKNOWN
+            for k in ex_ops_pb2.OperationResponseMapping.keys()[1:]
```

Comment:
Is this always sorted?

Reply:
I don't believe that's a safe assumption for us to make. Good catch!